msal/src/main/java/com/microsoft/identity/client/NativeAuthPublicClientApplication.kt

```diff
@@ -0,0 +1,111 @@
+import com.microsoft.identity.client.statemachine.InvalidAttributesError
+import com.microsoft.identity.client.statemachine.InvalidEmailError
+import com.microsoft.identity.client.statemachine.InvalidPasswordError
+import com.microsoft.identity.client.statemachine.PasswordIncorrectError
+import com.microsoft.identity.client.statemachine.UserAlreadyExistsError
+import com.microsoft.identity.client.statemachine.UserNotFoundError
+import com.microsoft.identity.common.crypto.AndroidAuthSdkStorageEncryptionManager
+import com.microsoft.identity.common.internal.cache.SharedPreferencesFileManager
+import com.microsoft.identity.common.internal.controllers.LocalMSALController
+import com.microsoft.identity.common.internal.controllers.NativeAuthController
+import com.microsoft.identity.common.internal.net.cache.HttpCache
+import com.microsoft.identity.common.java.authorities.Authority
+import com.microsoft.identity.common.java.cache.ICacheRecord
+import com.microsoft.identity.common.java.commands.CommandCallback
+import com.microsoft.identity.common.java.controllers.CommandDispatcher
+import com.microsoft.identity.common.java.controllers.results.ICommandResult
+import com.microsoft.identity.common.java.eststelemetry.PublicApiId
+import com.microsoft.identity.common.java.exception.BaseException
+import com.microsoft.identity.common.java.logging.LogSession
+import com.microsoft.identity.common.java.logging.Logger.LogLevel
+import com.microsoft.identity.common.java.providers.microsoft.azureactivedirectory.AzureActiveDirectory
+import com.microsoft.identity.common.java.util.ResultFuture
+import com.microsoft.identity.common.java.util.checkAndWrapCommandResultType
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.SupervisorJob
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+
+class NativeAuthPublicClientApplication(
+    private val nativeAuthConfig: NativeAuthPublicClientApplicationConfiguration
+) : INativeAuthPublicClientApplication, PublicClientApplication(nativeAuthConfig) {
+
+    private lateinit var sharedPreferencesFileManager: SharedPreferencesFileManager
+
+    init {
+        initializeApplication()
+        initializeSharedPreferenceFileManager(nativeAuthConfig.appContext)
+    }
+
+    companion object {
+        /**
+         * Name of the shared preference cache for storing NativeAuthPublicClientApplication data.
+         */
+        private const val NATIVE_AUTH_CREDENTIAL_SHARED_PREFERENCES =
+            "com.microsoft.identity.client.native_auth_credential_cache"
+
+        internal val TAG = NativeAuthPublicClientApplication::class.java.toString()
+
+        val pcaScope = CoroutineScope(SupervisorJob() + Dispatchers.IO)
+    }
+
+    @Throws(MsalClientException::class)
+    private fun initializeApplication() {
+        val context = nativeAuthConfig.appContext
+
+        setupTelemetry(context, nativeAuthConfig)
```

Comment:
Do we have a conclusion on telemetry?

Reply:
@SammyO

Reply:
We should remove this, as we're not doing anything with this telemetry right now. We'll add this or something like this to our public preview release.
@SaurabhMSFT  you can remove this call.